File: project-{topic}-optimized/app/models/product.py
import datetime
from app.database import db
from sqlalchemy import Column, Integer, String, Text, Numeric, DateTime, Boolean, ForeignKey
from sqlalchemy.orm import relationship

class Product(db.Model):
    __tablename__ = 'products'
    id = Column(Integer, primary_key=True)
    name = Column(String(255), nullable=False, index=True)
    slug = Column(String(255), unique=True, nullable=False)
    description = Column(Text, nullable=True)
    price = Column(Numeric(10, 2), nullable=False)
    stock_quantity = Column(Integer, nullable=False, default=0)
    image_url = Column(String(255), nullable=True)
    is_active = Column(Boolean, default=True)
    created_at = Column(DateTime, default=datetime.datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.datetime.utcnow, onupdate=datetime.datetime.utcnow)

    category_id = Column(Integer, ForeignKey('categories.id'), nullable=False)
    # category relationship is in Category model via backref

    def __repr__(self):
        return f'<Product {self.name}>'

File: project-{topic}-advanced/frontend/src/types/product.ts
// Matching backend Product type
export interface Product {
  id: string;
  name: string;
  description: string;
  price: number;
  stock: number;
  imageUrl?: string | null;
  categoryId: string;
  category?: { // Include nested category for display
    id: string;
    name: string;
  };
  createdAt: string;
  updatedAt: string;
}